File: CSharp/SpecStudioExamplesTestsCSharp/production/CatalogItem.cs
namespace production
{
    public class CatalogItem
    {
        public SimpleText Name { get; }
        public Dollar Price { get; }

        public CatalogItem(SimpleText name, Dollar price)
        {
            Name = name;
            Price = price;
        }

        public class Builder
        {
            private SimpleText _name;
            private Dollar _price;

            public Builder Name(SimpleText value) { _name = value; return this; }
            public Builder Price(Dollar value) { _price = value; return this; }

            public CatalogItem Build() =>
                new CatalogItem(_name, _price);
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/main/java/production/CatalogItem.java
package production;

import java.util.Objects;

public class CatalogItem {
    public final SimpleText name;
    public final Dollar price;

    public CatalogItem(SimpleText name, Dollar price) {
        this.name = name;
        this.price = price;
    }

    public static class Builder {
        private SimpleText name;
        private Dollar price;

        public Builder name(SimpleText name) { this.name = name; return this; }
        public Builder price(Dollar price) { this.price = price; return this; }

        public CatalogItem build() {
            return new CatalogItem(name, price);
        }
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof CatalogItem)) return false;
        CatalogItem that = (CatalogItem) o;
        return Objects.equals(name, that.name)
            && Objects.equals(price, that.price);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, price);
    }

    @Override
    public String toString() {
        return "CatalogItem{" + "Name=" + name + ", " + "Price=" + price + "}";
    }
}
